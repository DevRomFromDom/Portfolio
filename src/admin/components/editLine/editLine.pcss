.edit-line-component {
  &.blocked {
    opacity: 0.5;
    filter: grayscale(100%);
    pointer-events: none;
  }
}

.title {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.input {
  width: 80%;
}
.icons{
  width: 10.3%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.buttons {
  display: flex;
  padding: 1px;
}

.button-icon {
  margin-right: 20px;

  &:last-child {
    margin-right: 0;
  }
}