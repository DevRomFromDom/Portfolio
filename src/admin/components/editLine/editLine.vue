<template>
    <div class="edit-line-component" :class="{ blocked: blocked }">
        <div class="title" v-if="editmode === false">
            <div class="text">{{ value }}</div>
            <div class="icons">
                <icon symbol="pencil" grayscale @click="editmode = true"></icon>
                <icon grayscale symbol="trash"  @click="$emit('removeCategory')"/>
            </div>
        </div>
        <div v-else class="title">
            <div class="input">
                <app-input
                    placeholder="Название новой группы"
                    :value="value"
                    :errorMessage="errorText"
                    @input="$emit('input', $event)"
                    @keydown.native.enter="onApprove"
                    autofocus="autofocus"
                    no-side-paddings="no-side-paddings"
                ></app-input>
            </div>
            <div class="buttons">
                <div class="button-icon">
                    <icon symbol="tick" @click="onApprove"></icon>
                </div>
                <div class="button-icon">
                    <icon symbol="cross" @click="noChange"></icon>
                </div>
            </div>
        </div>
    </div>
</template>

<script>
export default {
    props: {
        value: {
            type: String,
            default: "",
        },
        errorText: {
            type: String,
            default: "",
        },
        editModeByDefault: Boolean,
        blocked: Boolean,
    },
    data() {
        return {
            editmode: this.editModeByDefault,
            title: this.value,
        };
    },
    methods: {
        onApprove() {
            if (
                this.title.trim() === this.value.trim() &&
                this.value.trim() !== ""
            ) {
                this.editmode = false;
            } else {
                if (this.value.trim() !== "") {
                    this.$emit("approve", this.value);
                    this.title = this.value
                    this.editmode = false;
                } else {
                    this.$emit("approve", this.value);
                }
            }
        },
        noChange(){
            this.editmode = false;
            this.title = this.value
        }
    },
    components: {
        icon: () => import("components/icon"),
        appInput: () => import("components/input"),
    },
};
</script>

<style lang="postcss" scoped src="./editLine.pcss">

</style>
