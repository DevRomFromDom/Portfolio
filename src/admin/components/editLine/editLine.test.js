import editLine from "./editLine.vue"
import {mount} from "@vue/test-utils"
import { expect } from "@jest/globals"




it("EditLine has been mounted", async ()=>{
    const wrapper = mount(editLine)
    expect(wrapper.element).toMatchSnapshot();
    await wrapper.find(".edit").trigger("click")
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find(".buttons").exists()).toBe(true)
    await wrapper.find(".noChange").trigger("click")
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find(".icons").exists()).toBe(true)

})