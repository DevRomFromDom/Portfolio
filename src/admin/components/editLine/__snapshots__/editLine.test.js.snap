// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditLine has been mounted 1`] = `
<div
  class="edit-line-component"
>
  <div
    class="title"
  >
    <div
      class="text"
    >
      
    </div>
     
    <div
      class="icons"
    >
      <button
        class="edit icon-component is-pencil grayscale no-words"
        data-text=""
        type="button"
      />
       
      <button
        class="icon-component is-trash grayscale no-words"
        data-text=""
        type="button"
      />
    </div>
  </div>
</div>
`;

exports[`EditLine has been mounted 2`] = `
<div
  class="edit-line-component"
>
  <div
    class="title"
  >
    <div
      class="input"
    >
      <label
        class="input no-side-paddings"
      >
        <!---->
         
        <input
          autofocus="autofocus"
          class="input__elem field__elem"
          placeholder="Название новой группы"
        />
         
        <div
          class="input__error-tooltip"
        >
          <div
            class="input__error-tooltip-container"
          >
            
          </div>
        </div>
      </label>
    </div>
     
    <div
      class="buttons"
    >
      <div
        class="button-icon"
      >
        <button
          class="icon-component is-tick no-words"
          data-text=""
          type="button"
        />
      </div>
       
      <div
        class="button-icon"
      >
        <button
          class="noChange icon-component is-cross no-words"
          data-text=""
          type="button"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`EditLine has been mounted 3`] = `
<div
  class="edit-line-component"
>
  <div
    class="title"
  >
    <div
      class="text"
    >
      
    </div>
     
    <div
      class="icons"
    >
      <button
        class="edit icon-component is-pencil grayscale no-words"
        data-text=""
        type="button"
      />
       
      <button
        class="icon-component is-trash grayscale no-words"
        data-text=""
        type="button"
      />
    </div>
  </div>
</div>
`;
